packages/indexer/src/sinks/csv.ts

```diff
@@ -0,0 +1,95 @@
+import type { Cursor } from "@apibara/protocol";
+import { Sink, type SinkWriteArgs } from "../sink";
+import { type Stringifier, stringify, type Options } from "csv-stringify";
+import fs from "node:fs";
+
+export type CsvArgs = {
+  /**
+   * csv-stringy options
+   * @reference https://csv.js.org/stringify/options/
+   */
+  csvOptions?: Options;
+  /**
+   * filepath for your csv file
+   */
+  filepath: string;
+};
+
+export type CsvSinkOptions = {
+  /**
+   * An optional column name used to store the cursor value. If specified,
+   * the value of this column must match the `endCursor.orderKey` for each row.
+   */
+  cursorColumn?: string;
+};
+
+class CsvSink<TData extends Record<string, unknown>> extends Sink<TData> {
+  private _config: CsvSinkOptions;
+  private _stringifier: Stringifier;
+
+  constructor(stringifier: Stringifier, config: CsvSinkOptions) {
```

Comment:
```suggestion
  constructor(private _stringifier: Stringifier, private _config: CsvSinkOptions) {
```

You can assign properties directly in the constructor.

Reply:
See [this article](https://kendaleiv.com/typescript-constructor-assignment-public-and-private-keywords/) for an introduction.